spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -755,6 +755,26 @@ private static String sqlString(org.apache.iceberg.expressions.Literal<?> lit) {
     return null;
   }
 
+  /**
+   * Returns an Iceberg Table by its name from a Spark V2 Catalog
+   *
+   * @param spark SparkSession used for looking up catalog references and tables
+   * @param name  The multipart identifier of the Iceberg table
+   * @return an Iceberg table
+   */
+  public static org.apache.iceberg.Table loadIcebergTable(SparkSession spark, String name)
+      throws ParseException, NoSuchTableException {
+    CatalogAndIdentifier catalogAndIdentifier = catalogAndIdentifier(spark, name);
+
+    CatalogPlugin catalog = catalogAndIdentifier.catalog;
+    Preconditions.checkArgument(catalog instanceof BaseCatalog, "Catalog %s(%s) is not an Iceberg Catalog",
+        catalog.name(), catalog.getClass().toString());
+    BaseCatalog baseCatalog = (BaseCatalog) catalogAndIdentifier.catalog;
+
+    Table sparkTable = baseCatalog.loadTable(catalogAndIdentifier.identifier);
+    return toIcebergTable(sparkTable);
```

Comment:
Does it make sense to refresh the iceberg table here to make sure the table's `ops` up to date? cc @RussellSpitzer,  @kbendick, @szehon-ho

Reply:
Pro is that caller will always get an updated ops of the table.
Con is that `refresh` involves IO, which is not necessary if user is OK with the stale `ops`.

Reply:
Hi, I'm not sure I understand exactly, baseCatalog.loadTable returns a table with fresh ops, doesnt it?

Reply:
No, it may be cached, check https://github.com/apache/iceberg/blob/7a5802808a3091e4a32a11043c220d27a9505b86/core/src/main/java/org/apache/iceberg/CachingCatalog.java#L75

Reply:
I see, could be left to the user then (and note in javadoc)?  It seems its easy for caller to call refresh() if they want.

Reply:
Updated the Java doc.

Reply:
I think the update to the Java doc is sufficient.

Reply:
This all seems fine to me. I think we should merge this :) I actually would argue this should return exactly the Iceberg table that would be used by the underlying Spark table, so no refreshing allowed.